### ISSUE TITLE
Devices constantly become "unavailable" at Home Assistant for a few minites

### ISSUE THREAD

**Issue by SvenAbels** [petretiandrea/home-assistant-tapo-p100#687]
## Version of the integration
latest (v2.13.0)

## Configuration
Exactly 20 tapo devices: 18 of them are P100 and 2 of them are P110

## Describe the bug
All devices are connected to Home Assistant without any problems and are listing their current poweer consumption. But from time to time some of them become "unavailbale" at Home Assistant. This happens to multiple devices. After 1-5 minutes, the device is typically available again. However, the devices themselve have a proper web connection and I can still see their current valiues (including changes) in the tapo Android app. 

In the logs menu of Home Assistant, I can only see that the device becomes unavailable but without any further details about the reason.  The devices are connected to the tapo extension using their IP. The IP does not change and the device is still pingable from the PC where Home Assistant is installed on.

I already modified the refresh times of the devices as I was fearing that they might be too low. So I increased them from 10 to 20 seconds but this did not change anything.

The problem does not happen at all times: It sometimes works for many hours without any problem and then the problem appears again - e.g. 10 times an hour for a few hours...  I'm experiencing this problem since >4 weeks now (since I started using tapos).

**Comment by Elastiek** [comment]
Same problem here.
When I trie to add the device again in HASSIO it say's "device already configured.
Anyone with a solution ?

**Comment by SvenAbels** [comment]
For me, it works to reload the device by clicking on the 3 dots in the device overview:

![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/1993969/31d6089d-abd8-4b72-9654-e139859e8812)


However, this is not a good solution because it only works for a while and I cannot reload several devices a day...

**Comment by pan07gaz94** [comment]
Same problem here too.
11/11 devices not available.

**Comment by pan07gaz94** [comment]
> For me, it works to reload the device by clicking on the 3 dots in the device overview:
> 
> ![image](https://private-user-images.githubusercontent.com/1993969/298716030-31d6089d-abd8-4b72-9654-e139859e8812.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5OTk5MDYsIm5iZiI6MTcwNTk5OTYwNiwicGF0aCI6Ii8xOTkzOTY5LzI5ODcxNjAzMC0zMWQ2MDg5ZC1hYmQ4LTRiNzItOTY1NC1lMTM5ODU5ZTg4MTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMjNUMDg0NjQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzM5ZjA2YmYyZDk5OTJkNDFkMzkxMjM4NmE1MjFmYTAzMWNlNGQ5ZDM2NjBiOTExY2EyYjlhN2FhY2NhNWM2NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.xMXvPGrH6-HOoFQFuVDF37DEduMFg-L2SjHemHMDAsg)
> 
> However, this is not a good solution because it only works for a while and I cannot reload several devices a day...

For me it didn't working

**Comment by ChrisMarriott38** [comment]
I get this happening also. Constantly for some.

**Comment by JCummings49** [comment]
I get similar behavior with the P125M. The device just turns off periodically during the day. It does not come back on, it must be manually turned on. Very annoying.

**Comment by pan07gaz94** [comment]
i reconnect all my plugs into my integration but when i restart my HAOS i lost again all my tapo p100 entities

**Comment by uncle-jegan** [comment]
Are your devices accessible in the Tapo app when they "unavailable" in HA? I experimented with isolation of network and DNS records, and get same problem. But when DNS is 8.8.8.8 and network connected directly to internet (devices online in Tapo app) there is no problem with "unavailable" condition. 
Devices: 
tapo L530 v1.1.0 build 230721 rel.224802
tapo P115 v1.3.0 build 230905 rel.152200

**Comment by ChrisMarriott38** [comment]
My actual devices don't become unavailable at all, lights always green also on the sockets. you can see them all the time in the app and home assistant - it's just in the log file that there are overheat and power issues detected. Unless the reconnects are unnoticed. 

I mean this is silly because this is JUST today. Look how many flipping entries there are!!!!!
![Subwoofer](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/142601601/ec0850a5-52ee-4afa-a757-3f3b48e2ddb4)

I have noticed it seems to also VARY on which device does this. For example. I have 7 sockets currently linked up. It's only that device so far today reporting this. But the other day i had a similar one on another socket doing this.

**Comment by SvenAbels** [comment]
> Are your devices accessible in the Tapo app when they "unavailable" in HA? 

Yes my devices are still accessible via the tapo application but not via HomeAssistant during that time. I also get no power consumption values in home assistant during that time but I do the real time power consumption in die App.

**Comment by ChrisMarriott38** [comment]
All mine are fine in Tapo, nothing is visible as changing. I don't even see any devices states change in home assistant. The only real clue i think is the log book saying something weird happening. Feels like the integration is causing it.

**Comment by ChrisMarriott38** [comment]
I have moved a Wifi repeater next to one of the plugs which this happened frequently on to diagnose if its a signal strength issue. Will report back.

**Comment by ChrisMarriott38** [comment]
> Hi,
> 
> for me the signal is not the problem. My repeater is in the same room for 4 of the Tapos but they still got this problem from time to time. Also, as I described earlier, I can ping the devices during the unavialability time and their real time data i also visible in the tapo app.

Agree, i don't think it will likely be to do with this. I'm just going to make certain with this test. 

Are your devices on 2.4 or 5ghz frequency? That can also be known to be an issue.

**Comment by SvenAbels** [comment]
My devices are all on 2.4 Ghz

**Comment by SvenAbels** [comment]
Thanks for the hint. I checked it and the checkbox is already deactiavted on my devices.

I also notices, that the unavailability happens very infrequently: Sometimes I get then 3-4 days in a row every few hours and then sometimes I get no problems for a week or two..

**Comment by ChrisMarriott38** [comment]
Sadly, use Google DNS, and i have not used the dynamic track part as i already had fixed IP's for each device. 

Yea same to the above, its so random and sporadic, sometimes is super frequent. 

I did wonder if it could be to do with the device refresh rate. i think mine are default 30s or so.

**Comment by whoisthedaddy** [comment]
Ok thanks will keep an eye out and comment further - the only two things I’ve changed today are the DNS to 8.8.8.8 and the checkbox.   My refresh rate is also set to 30 I believe…

**Comment by ChrisMarriott38** [comment]
Anyone tried any older versions? To see if its a more recent issue?

**Comment by whoisthedaddy** [comment]
Nope, I've not tried going back to an older version - wasn't sure whether I'd have to re-add everything which is far too painful!  All the re-pairing, adding everything back in and then updating all the entities!

**Comment by ChrisMarriott38** [comment]
We've not heard anything in regards to this reply. I've updated to the latest home assistant and actually noticed that the TP-Link official integration now detects Tapo devices. It's still early on i think it said so maybe buys? It doesnt detect all the entities like this Tapo integration though. 

I can see 4 entities per device on the TP-Link . Switch, current consumption, todays consumption, total consumption. 

On this Tapo one i can see 8 entities per device. Switch, Current power, todays energy, Month energy, (These are extras - month run time, signal level, todays run time. overheat.)

**Comment by ChrisMarriott38** [comment]
So i tried disable a device from the TAPO integration. Just one of the plugs that reported the most log book errors. And added it via the new TP-LINK Direct integration but i think i get the same log book errors. It seems strange. Unsure if its related to the integrations now or actually if it's more related to the HA build. However, i looked back at my older test HA version. 

Home Assistant 2023.8.2
Frontend 20230802.0 - latest

And i see the same Logbook issues in that also.

**Comment by ChrisMarriott38** [comment]
Anyone see this or similar in their logs?
![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/142601601/0ad6409c-2b70-49b4-beae-5599ea2d1413)

![image](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/142601601/f4f19c08-903c-4a22-b0a2-a79fabb3dabb)

**Comment by SvenAbels** [comment]
I did not have this experience for several weeks now. Thus I intent to close this issue unless it's still unresolved for others...

**Comment by devangapatel** [comment]
I am seeing the same issue/pattern with the Tapo Mini Smart Wifi Plug TP 15. Wifi strength is not a concern. Also the device works fine in their native Tapo app when it is unavailable in HA. The older (4+year old) TP-Link smart plugs do not have this issue but the newer Tapo TP15 is experiencing this problem. This is definitely a bug with HA integration. Can someone please help fix this?

**Comment by jaymunro** [comment]
I have this issue. Checking to find out when it started...

**Comment by jaymunro** [comment]
It started before the log pruning date 10 days ago, so no info there. I have 7 devices and the issue is present on all of them with unavailability occurring several times an hour and lasting 5-40 minutes.

I am seeing a lot of these in the log:
```
This error originated from a custom integration.

Logger: custom_components.tapo.coordinators
Source: helpers/update_coordinator.py:347
integration: TP-Link Tapo (documentation, issues)
First occurred: 11:34:17 AM (28 occurrences)
Last logged: 12:07:26 PM

Error fetching tapo data: Unexpected exception:
```
Note HA was restarted at 11:33am

I have enabled debug logging to try and get more detail.

**Comment by jaymunro** [comment]
Not seeing anything more helpful in the DEBUG entries:
```
2024-04-23 12:17:35.803 ERROR (MainThread) [custom_components.tapo.coordinators] Error fetching tapo data: Unexpected exception: 
2024-04-23 12:17:35.803 DEBUG (MainThread) [custom_components.tapo.coordinators] Finished fetching tapo data in 10.003 seconds (success: False)
```

**Comment by elliotcz97** [comment]
Same Issue still happens, for me it's a bit better because it reconnects very fast, but still annoying.

**Comment by LaszloV369** [comment]
Hello everyone. I have found a solution, at least for me it works. I have 10 tapo devices one or two became unvailable randomly, very annoying!!! I have observed that the wifi channel on the router was set to Auto. I mooved it to manual setting and set to be always on channel 6. I chose this cahannel using wifi analyzer android app on the phone to se wich channel is the most "empty" - no neighbouring wifis to interfere. Since than I hand NO MORE unavailability issues on any of the devices! My router is a Tplink Archer x73. How this helps!

**Comment by hareenl** [comment]
I'm also seeing this issue and it happens constantly through out the day accross 11 Tapo P110 outlets. It sometimes happens to groups of outlets and only certain outlets at other times. I have strong WiFi across the house with 3 access points on channels 1,6 and 11. All devices on the network are pinged utilizing uptime kuma and nothing drops on the network itself. The following was seen on the HA logs.

![Screenshot 2024-06-26 at 1 16 46 PM](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/128954351/65fbd9ba-9a12-4156-82e0-71dcf26eac23)
![Screenshot 2024-06-26 at 1 16 56 PM](https://github.com/petretiandrea/home-assistant-tapo-p100/assets/128954351/808f6262-540c-44ef-b124-766c3d1a5f70)

**Comment by jaymunro** [comment]
It used to happen to me a lot. Then I completely uninstalled the HACS version - deleted all entities and deleted the HACS integration version. That was months ago and it's only happened a couple of times since, usually after a significant upgrade in OS or HA. Suggest you try that or say that your only running on the HA Core integration and have removed the HACS one (which doesn't seem to be getting any love).

**Comment by andyevans-eng** [comment]
Since adding a Tapo P110 to my network one of my Kasa KP115 plugs occasionally becomes unavailable, but it was rock solid before, and the P110 becomes unavailable every 5 minutes or so. Signal strength is excellent. Only started having problems when P110 was set up. While devices are unavailable, they are still online in the Tapo app. It is getting very frustrating as my energy data in HA is intermittent because of it being unavailable, but all data in the Tapo app

**Comment by zqnikals** [comment]
same issue. I have 11 plugs for a few weeks. the signal is strong. today they became to become unavailable one by one. Also, i found One strange thing: sometimes plugs set auto-off timer to 120. I have never had such automation. 

![image](https://github.com/user-attachments/assets/eb93fda9-2bc5-4ebc-991c-028e3b77d17e)

**Comment by Kopetan4egX** [comment]
> sometimes plugs set auto-off timer to 120. I have never had such automation.

Auto-off timer time (120 by default for me, and seems like for you too) and the timer (meaning on/off status) itself are separate entities. The one with seconds just shows the time before turning of, it won't turn off the socket, if the entity  "Auto-off enabled" is off/

**Comment by s55ma** [comment]
Also the same problem, but with Tapo P115 plugs. Kasa plugs works fine and other non Tplink devices also work fine.

**Comment by c-kick** [comment]
This is still an issue as of version 2024.8.3. And still, the only workaround to get things working again is blocking internet access to all the P115 plugs. But that doesn't even restore everything, as some of the device entities will still not work (such as current and total power usage, which is one of the main reasons I invested in these plugs).

**Comment by rheeqp** [comment]
This is actually major issue, because it will reset changed time and updated time. I hope they will proritize to fix this issue on the next update.

**Comment by cassiomolin** [comment]
I noticed the same issue with the Tapo T315 temperature and humidity sensor.
I use the T315 sensor in the [generic hygrostat](https://www.home-assistant.io/integrations/generic_hygrostat/). When the sensor becomes unavailable, the dehumidifier turns off (regardless whether `sensor_stale_duration` is present or not).

**Comment by LaszloV369** [comment]
What solved this issue in my case was separating 2.4G wifi from 5G wifi. Made separate SSID for 5Ghz and all devices now remained on the 2.4G network. Since the separation i have almoast 0 disconnections. Hope this helps. !!! :)

**Comment by dako5004** [comment]
I have this issue, too, and I have a suspicious behavior: The issue appears when I switch on my PC. I use the PING integration of HA in order to switch on my printer when I turn on the PC. So, I deactivated ping for my PC and the problems have disappeared. 
As others too, I have several access points and the RPi with HA on it is connected via cable to one of these access points. What I see is that HA gets unavailable as soon as I switch on my PC, gets available, unavailable and so on. Hence, for me, it seems not obviously a problem of my TAPO devices but I think the PING integration is suspicious as the pings seem to be blocked (and thus HA) by the router.

In past, I watched suspicious behavior of my wife's Android mobile phone: Although Wifi was enabled on it and Wifi was available at home, the device did not connect. I was detecting its presence via ping integraation as well. Now I use the sensors of the mobile HA app for presence detection.

I will look for that further. Is there anyone else both using ping integration and having problems on Tapo devices?

**Comment by gnr8shn** [comment]
Similar to above. P100, P110's. Toggle off and on roughly every 10s.

**Comment by 36below** [comment]
I am seeing this from "ping -O":
`no answer yet for icmp_seq=10108
no answer yet for icmp_seq=10109
no answer yet for icmp_seq=10110
64 bytes from y.y.y.y: icmp_seq=10111 ttl=254 time=16.6 ms
no answer yet for icmp_seq=10112
64 bytes from y.y.y.y: icmp_seq=10113 ttl=254 time=9.32 ms
64 bytes from y.y.y.y: icmp_seq=10114 ttl=254 time=12.8 ms
64 bytes from y.y.y.y: icmp_seq=10115 ttl=254 time=8.39 ms
64 bytes from y.y.y.y: icmp_seq=10116 ttl=254 time=6.52 ms
64 bytes from y.y.y.y: icmp_seq=10117 ttl=254 time=9.02 ms
64 bytes from y.y.y.y: icmp_seq=10118 ttl=254 time=79.5 ms
64 bytes from y.y.y.y: icmp_seq=10119 ttl=254 time=3.20 ms
64 bytes from y.y.y.y: icmp_seq=10120 ttl=254 time=2.71 ms
64 bytes from y.y.y.y: icmp_seq=10121 ttl=254 time=2.61 ms
64 bytes from y.y.y.y: icmp_seq=10122 ttl=254 time=6.82 ms
64 bytes from y.y.y.y: icmp_seq=10123 ttl=254 time=18.2 ms
64 bytes from y.y.y.y: icmp_seq=10124 ttl=254 time=3.25 ms
64 bytes from y.y.y.y: icmp_seq=10125 ttl=254 time=4.94 ms
64 bytes from y.y.y.y: icmp_seq=10126 ttl=254 time=3.41 ms
no answer yet for icmp_seq=10127
no answer yet for icmp_seq=10128
no answer yet for icmp_seq=10129
no answer yet for icmp_seq=10130
no answer yet for icmp_seq=10131
no answer yet for icmp_seq=10132
no answer yet for icmp_seq=10133
no answer yet for icmp_seq=10134
no answer yet for icmp_seq=10135
no answer yet for icmp_seq=10136
no answer yet for icmp_seq=10137
From x.x.x.x icmp_seq=10138 Destination Host Unreachable
no answer yet for icmp_seq=10138
no answer yet for icmp_seq=10139
no answer yet for icmp_seq=10140
From x.x.x.x icmp_seq=10141 Destination Host Unreachable
no answer yet for icmp_seq=10141
no answer yet for icmp_seq=10142
no answer yet for icmp_seq=10143
no answer yet for icmp_seq=10144
From x.x.x.x icmp_seq=10145 Destination Host Unreachable
no answer yet for icmp_seq=10145
no answer yet for icmp_seq=10146
no answer yet for icmp_seq=10147
From x.x.x.x icmp_seq=10148 Destination Host Unreachable
no answer yet for icmp_seq=10148
no answer yet for icmp_seq=10149
no answer yet for icmp_seq=10150
From x.x.x.x icmp_seq=10151 Destination Host Unreachable
no answer yet for icmp_seq=10151
no answer yet for icmp_seq=10152
no answer yet for icmp_seq=10153
no answer yet for icmp_seq=10154
no answer yet for icmp_seq=10155
no answer yet for icmp_seq=10156
64 bytes from y.y.y.y: icmp_seq=10157 ttl=254 time=9.43 ms
64 bytes from y.y.y.y: icmp_seq=10158 ttl=254 time=3.63 ms
64 bytes from y.y.y.y: icmp_seq=10159 ttl=254 time=2.92 ms`

I don't think it's anything to do with HomeAssistant. I've moved the IP and no change. I've got another one at another location and it is absolutely fine.

**Comment by ahmetburakbarakli** [comment]
i had a similar issue with my tapo L530e going unavailable for short periods and then coming back. found this discussion while looking for a fix. after @dako5004's comment about the ping integration, i disabled the ping integration i was using to monitor my server, and the bulb has been stable since.
thanks for pointing it out.

![Image](https://github.com/user-attachments/assets/5834e376-b481-4507-bbb9-290e28808501)

**Comment by noodle-** [comment]
Maybe this will help someone; I had similar issues and then moved to the official TP-link integration instead: https://www.home-assistant.io/integrations/tplink/
Haven't had this issue anymore since then.

**Comment by elliotcz97** [comment]
I also moved to the official integration! But for me unfortunately it still happens!
I ended up making an automation which notifies me if the device became unavailable so I can powercycle (unplug replug) to resolve the issue for approximately a week.

EDIT: I’m starting to think that it’s a firmware issue, because when this happens then it’s also not available in homebridge nor in the official Tapo app!
